### PR TITLE
go: update to 1.19, add tests

### PR DESCRIPTION
**Pull request by subnut**
Supersedes #38474
Supersedes #38385

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686-glibc

### CODE REVIEW

**Comment by paper42** [comment]
this PR seems to be in the best shape, cc @the-maldridge

File: srcpkgs/go/template [hunk 1]
```diff
@@ -31,11 +32,37 @@ if [ "$CROSS_BUILD" ]; then
...
+	fi
+	if [ "$XBPS_LIBC" = musl ]; then
+		patches+=" fix-musl-tests.patch"
+	fi
+	for patch in $patches; do
+		cp "$FILESDIR/$patch" "$wrksrc/$patch"
+	done
+	for patch in $patches; do
+		patch -Np1 <"$patch"
+	done
```
Comment (paper42) [inline srcpkgs/go/template:54]: this is ugly and the tests that fail on i686 probably also fail on other 32bit archs, I would just patch all of them out unconditionally.

**Comment by the-maldridge** [comment]
@subnut this looks fine with the exception of the tests.  Please apply the requested changes, then good for merge.

File: srcpkgs/go/template (hunk 1 above)
Comment (subnut) [reply srcpkgs/go/template]: Done

**Comment by subnut** [comment]
@the-maldridge Done.

**Pull request merged**